ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java

```diff
@@ -411,4 +419,17 @@ public void clearCookies(com.facebook.react.bridge.Callback callback) {
     return getReactApplicationContext()
         .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class);
   }
+
+  /**
+   * prefetch image to the fresco image disk cache
+   * @param url
+   */
+  @ReactMethod
+  public void prefetchImage(String url) {
+    ImageRequest request = ImageRequestBuilder.newBuilderWithSource(Uri.parse(url)).build();
+    ImagePipeline pipeline = Fresco.getImagePipeline();
+    if(!pipeline.isInDiskCache(request).getResult()) {
```

Comment:
You don't need to do this check, which as @mkonicek said, isn't correct anyways. The pipeline won't go out on network unless the image isn't in cache. You can omit the if entirely and then don't have to worry about DataSources.